homeassistant/components/tuya/event.py

```diff
@@ -120,14 +127,14 @@ def __init__(
         device: CustomerDevice,
         device_manager: Manager,
         description: EventEntityDescription,
+        dpcode_wrapper: DPCodeEnumWrapper,
     ) -> None:
         """Init Tuya event entity."""
         super().__init__(device, device_manager)
         self.entity_description = description
         self._attr_unique_id = f"{super().unique_id}{description.key}"
-
-        if dpcode := find_dpcode(self.device, description.key, dptype=DPType.ENUM):
-            self._attr_event_types: list[str] = dpcode.range
+        self._dpcode_wrapper = dpcode_wrapper
+        self._attr_event_types = dpcode_wrapper.enum_type_information.range
```

Comment:
Note: this needs to be changed to `type_information` if #156039 is accepted and merged.
```python
        self._attr_event_types = dpcode_wrapper.enum_type_information.range
        # vs
        self._attr_event_types = dpcode_wrapper.type_information.range
```